auth/mtls/common.go

```diff
@@ -20,21 +20,48 @@ import (
 	"crypto/x509"
 	"fmt"
 	"os"
+	"path/filepath"
 )
 
 // LoadRootOfTrust will load an CA root of trust(s) from the given
 // file and return a CertPool to use in validating certificates.
 // All CA's to validate against must be presented together in the PEM
 // file.
-func LoadRootOfTrust(filename string) (*x509.CertPool, error) {
-	// Read in the root of trust for client identities
-	ca, err := os.ReadFile(filename)
+// If the file is a directory, LoadRootOfTrust will load all files
+// in that directory.
+func LoadRootOfTrust(path string) (*x509.CertPool, error) {
+	capool := x509.NewCertPool()
+
+	loadRoot := func(filename string) error {
```

Comment:
optional: feel free to ignore

you could structure this a bit differently and avoid the need for the function var

```
var certfiles []string
fi, err := os.Stat(path)
if err != nil { return ... }

if fi.IsDir() {
 files, err := os.ReadDir(path)
 if err != nil { return .. }
 for _, f := range files {
    certfiles = append(certfiles, filepath.join(path, f.Name())
 }
} else {
 certfiles = []string{path}
}

pool := x509.NewCertPool()
for _, file := range certfiles {
   // read file
  // append to pool
}
```

Reply:
I like it, it's simpler. Changed.